graph-query-ir/src/main/java/oracle/pgql/lang/ir/QueryPath.java

```diff
@@ -175,10 +193,20 @@ public String toString() {
   }
 
   private String printVariableLengthPathPattern(PathFindingGoal goal) {
-    String kValueAsString = kValue > 1 ? "TOP " + kValue + " " : "";
-    String allAsString = withTies ? "ALL " : "";
-    String goalAsString = goal == PathFindingGoal.REACHES ? "ANY" : goal.toString();
-    String result = kValueAsString + allAsString + goalAsString + " " + getSrc() + " ";
+    String pathPatternPrefix;
+    if (withTies) {
+      pathPatternPrefix = "ALL " + goal.toString() + " ";
+    } else if (goal == PathFindingGoal.REACHES) {
+      pathPatternPrefix = "ANY ";
+    } else if (kValue == 1) {
+      pathPatternPrefix = "ANY " + goal.toString() + " ";
+    } else {
+      String kValueAsString = kValue > 1 ? kValue + " " : "";
+      pathPatternPrefix = goal.toString() + " " + kValueAsString;
+    }
+
+    pathPatternPrefix += pathMode == PathMode.WALK ? "" : pathMode.toString() + " ";
```

Comment:
So path mode will also be allowed together with cheapest goal?

Reply:
in the SQL standard: yes
but current PGQL parser does not support it yet and the error will not be very nice when you try to parse e.g. ANY CHEAPEST ACYCLIC... something I'll try to improve.

Reply:
sorry, this part of the parser code indeed supports it already. But it's not enabled yet at the grammar level so you would get a parse error before it reaches this part of the code. Now it just needs to be enabled at the grammar level.